### PR TITLE
bpf: do not invoke llc from Makefiles

### PR DESCRIPTION
**Pull request by lmb**
bpf: do not invoke llc separately

    Update the various bpf Makefiles to not invoke llc anymore. This matches 
    what the loader does now.

    Signed-off-by: Lorenz Bauer <lmb@isovalent.com>

tools/dev-doctor: do not check for llc

    Compiling bpf doesn't invoke llc directly anymore, stop checking for its
    presence.

    Signed-off-by: Lorenz Bauer <lmb@isovalent.com>

### CODE REVIEW

File: bpf/tests/Makefile [hunk 1]
```diff
@@ -6,11 +6,10 @@ include ../../Makefile.defs
 MAKEFLAGS += -r
 
 CLANG ?= clang
-LLC   ?= llc
 
 FLAGS := -I$(ROOT_DIR)/bpf/include -I$(ROOT_DIR)/bpf -D__NR_CPUS__=$(shell nproc --all) -O2 -g
 
-CLANG_FLAGS := ${FLAGS} --target=bpf -std=gnu99 -nostdinc -emit-llvm
+CLANG_FLAGS := ${FLAGS} --target=bpf -std=gnu99 -nostdinc -march=bpf
```
Comment (borkmann) [inline bpf/tests/Makefile]: Is `-march=bpf` required in this case and covering sth which `--target=bpf` itself doesn't address?

**Review by joamaki — APPROVED** [review]

File: bpf/tests/Makefile (hunk 1 above)
Comment (lmb) [reply bpf/tests/Makefile]: Good point, we also don't pass this when compiling from Go. Removed.

**Review by borkmann — APPROVED** [review]

**Review by bimmlerd — APPROVED** [review]

**Review by dylandreimerink — APPROVED** [review]
LGTM

**Review by rgo3 — APPROVED** [review]

**Pull request merged**